### PR TITLE
Fix regression, bring back socket support, fixes #950

### LINKED ISSUE #950: `RedisClient(stream, options)` does not work anymore, reimplement it
`RedisClient(stream, options)`  does not work anymore. This ticket is here to re-implement socket support.

Proposal:

- `redis.createClient({socket:socket})`
- `option = clone(option);` will erase socket value, thus I will have to find another way.
- `create_stream` should then be renamed or a least take into account `options.socket`

[edit] the issue was not about `redis.createClient(unix_socket, options)`  but about `RedisClient(stream, options)`

### PR DESCRIPTION
**Pull request by FGRibreau**
Fix regression, bring back socket support, fixes #950

### CODE REVIEW

**Comment by FGRibreau** [comment]
Before this PR on master:

```
1425 passing (2m)
8 pending
10 failing
```

after this PR:

```
1425 passing (2m)
8 pending
10 failing
```

no regression were introduced @BridgeAR

**Comment by FGRibreau** [comment]
@BridgeAR https://coveralls.io/builds/4677286/source?filename=index.js#L1302 is not tested, but I will wait for your feedback. Should I implement `createClient(socket, options)` (which was not implemented before) or only keep the constructor implementation?

**Comment by FGRibreau** [comment]
Hello @BridgeAR, any news on this? :)

**Comment by BridgeAR** [comment]
@FGRibreau I'm really busy right now and I guess I won't find time before next monday :/

If you attend the [FOSDEM](https://fosdem.org/2016/) we could meet there though.

**Comment by FGRibreau** [comment]
Hello @BridgeAR,

This PR is too old so I will need to rebase, can you confirm you will be able to merge it then? I will only keep the constructor and remove the non-existing `createClient(socket, options)` :)

File: index.js [hunk 1]
```diff
@@ -33,9 +33,21 @@ function handle_detect_buffers_reply (reply, command, buffer_args) {
 
 exports.debug_mode = /\bredis\b/i.test(process.env.NODE_DEBUG);
 
-function RedisClient (options) {
+function RedisClient (globalOptions) {
```
Comment (FGRibreau) [inline index.js]: function RedisClient (options, stream) {

File: index.js [hunk 2]
```diff
@@ -33,9 +33,21 @@ function handle_detect_buffers_reply (reply, command, buffer_args) {
...
+
+    // ensure clone won't try to clone a socket
+    globalOptions.socket = null;
+
     // Copy the options so they are not mutated
-    options = clone(options);
+    var options = clone(globalOptions);
+
+    // add socket if present
+    options.socket = socket;
```
Comment (FGRibreau) [inline index.js]:
```js
if(stream){
this.stream = stream;
}
```

**Pull request closed without merging**